core/src/main/java/fr/sncf/osrd/envelope/OverlayEnvelopeBuilder.java

```diff
@@ -53,11 +57,13 @@ public OverlayEnvelopePartBuilder startDiscontinuousOverlay(
             EnvelopePartMeta meta,
             double initialSpeed
     ) {
+        sliceBaseEnvelope(cursor.getPartIndex(), cursor.getStepIndex(), cursor.getPosition(), cursor.getSpeed());
```

Comment:
Why are these two slices necessary ? Was it a mistake we didn't see before ?

Reply:
not at all, this used to be done once the envelope part is built, that's why this same PR removes plenty of fields from OverlayEnvelopePartBuilder: these were used to pinpoint where the envelope part started once it's built